### ISSUE TITLE
Bug in Luhn method passing all tests - TypeScript

### ISSUE THREAD

**Issue by MauricioRobayo** [exercism/problem-specifications#1670]
I have the following implementation for the Luhn algorithm:

```ts
class Luhn {
  static valid(number: string): boolean {
    
    const numberWithoutSpaces = number.replace(/ /g, '')
    
    if (!/^\d{2,}$/.test(numberWithoutSpaces)) {
      return false
    }
    
    let result = Number(numberWithoutSpaces[0])
    for (let i = numberWithoutSpaces.length - 2; i >= 0; i -= 2) {
      result += Number(numberWithoutSpaces[i + 1])
      const doubledNumber = Number(numberWithoutSpaces[i]) * 2
      result += doubledNumber > 9 ? doubledNumber - 9 : doubledNumber
    }
    
    return result % 10 === 0
  }
}

export default Luhn
```
This passes all the tests, but if the number length is even, it will be adding two times the first number, so it is wrong.

A test like this should catch the issue:

```js
expect(Luhn.valid("91")).toBeTruthy()
```

**Comment by kotp** [comment]
I am closing this in preference to where the problem specifications are located.  Please see exercism/problem-specifications#1670

**Comment by ErikSchierboom** [comment]
@MauricioRobayo if I pasted in the above code and run it with `Luhn.valid('91')`, it returns `false`. I'm slightly confused whether that is correct.

Since this issue has been created, two new test cases were added:

- https://github.com/exercism/problem-specifications/commit/d87fbbc5276b5230190365fca86f1675e793fa7b#diff-0a90142821ce891e157f2a453ff13353114485bed643723347518393c002a443
- https://github.com/exercism/problem-specifications/commit/306d5973eb32e4c84f629cc7fd8fc20874d1621e#diff-0a90142821ce891e157f2a453ff13353114485bed643723347518393c002a443

Do any of these catch the above bug?
